Lenasterg patch pdf. Uses config parameters in order to define the styling of pdf

lenasterg opened the pull request:
This patch adds some flexibility to the administrator in formatting the output pdf for a participant's  responses.The patch introduces 4 new config parameters for the pdf export, which are used to format the questions and responses cells. 

- $config['pdfquestionfill']:  	defines if the background in questions should be painted (1) or transparent (0)	
 - $config['pdfquestionbold'] Questions in bold (1) or normal (0)
 - $config['pdfquestionborder']  Border in questions. Accepts 0:no border, 1:border
 - $config['pdfresponseborder']  Border in responses. Accepts 0:no border, 1:border

LouisGac commented:
that seems ok to me.

Aestu commented on application/libraries/admin/pdf.php:
```diff
@@ -767,6 +767,26 @@ function addGidAnswer($sGroupName, $sGroupDescription, $bAllowBreakPage=false)
+  	}
+	
+  	$yiiresponseborder = Yii::app()->getConfig('pdfresponseborder');
+  	if ($yiiquestionborder=='0') 
```
Error? yiiquestionborder or yiiresponseborder?

Aestu commented:
Thanks a lot for your PR @lenasterg 

My comments:

- I tested functionality. It works fine except pdfresponseborder parameter. I don't see any change when I set it to 0. Take a look at my comment on code, I think there is an error.
- IMHO admins should be able to modify these new parameters from admin GUI in the same way as the existing ones.
- Take a look at [coding guidelines](http://manual.limesurvey.org/Coding_guidelines). The variable names don't follow them.
- Please, use spaces instead of tabs.

lenasterg commented:
@Aestu, I fixed the pdfResponseBorder (sorry about the initial bug). 
Also the variable names now follow the coling standards and I replaced the tabs with blanks. 
Added  necessary changes so admins can modify these new parameters from admin GUI in the same way as the existing ones.

Aestu commented:
@lenasterg : Thanks for the corrections. Functionality is fine. However, there are some problems yet:

Regarding variable names: To follow the standard, if yiiResponseBorde is a boolean the name should be bYiiResponseBorde. We know you can find existing variables that violate this rule but we try to improve.

Take a look at http://manual.limesurvey.org/Standard_for_Git_commit_messages. Perhaps creating a new PR with just one commit is better (sorry, I didn't realize this before my first comment)

lenasterg commented:
@aestu, sorry about not following the standard. I'm new in Github.
Is it better to  close this pull request and when I manage (or believe that) to comply with the standards to make another PR again?

SamMousa commented:
@aestu, I don't think we have Hungarion notation as a standard.

lenasterg commented:
@SamMousa It's in https://manual.limesurvey.org/Coding_guidelines#Naming_convention, but I forgot the prefix.

Aestu commented:
@lenasterg : We could accept your PR but @c-schmitz has the last say. He is the project leader.

Definitely, this patch is very usefull and we should accept it after you fix these minor problems. I suggested to generate a new PR to have just one commit and set a correct comment.

Shnoulle commented:
Looking at this PR : really think it's time to move the answer PDF to an external core plugin. Then any other dev can more easily hack it/improve it.

Aestu commented:
Perhaps, but pdf generation code is already very isolated in class pdf.php.

Anyway, in my opinion, this is not a reason for not accepting this patch and not having the functionality that @lenasterg has developed.

Aestu commented:
@lenasterg : We just talked about this PR in our team meeting. We agreed this can be accepted. Could you please make a new patch with just one commit with a comment following our standards? Do you need some help?

lenasterg commented:
@Aestu  OK. 
I'll close the PR and come back this a new one.

lenasterg commented:
New pull request is [456](https://github.com/LimeSurvey/LimeSurvey/pull/456)
